op-e2e/actions/proofs/isthmus_setcode_tx_test.go

```diff
@@ -189,3 +196,224 @@ func testInvalidSetCodeTxBatch(gt *testing.T, testCfg *helpers.TestCfg[any]) {
+	io.Reader
+	currPos uint
+}
+
+func (r *readerWithCurrPos) Read(p []byte) (n int, err error) {
+	n, err = r.Reader.Read(p)
+	r.currPos += uint(n)
+	return n, err
+}
+
+func (r *readerWithCurrPos) CurrPos() uint {
+	return r.currPos
+}
+
+func (r *readerWithCurrPos) ReadByte() (byte, error) {
+	var b [1]byte
+	_, err := r.Read(b[:])
+	return b[0], err
+}
+
+func runSetCodeTxTypeWithContractCreationBitSetTest(gt *testing.T, testCfg *helpers.TestCfg[any]) {
+	var (
+		aa = common.HexToAddress("0x000000000000000000000000000000000000aaaa")
+		bb = common.HexToAddress("0x000000000000000000000000000000000000bbbb")
+	)
+
+	t := actionsHelpers.NewDefaultTesting(gt)
+
+	// hardcoded because it's not available until after we need it
+	bobAddr := common.HexToAddress("0x14dC79964da2C08b23698B3D3cc7Ca32193d9955")
+
+	// Create 2 contracts, (1) writes 42 to slot 42, (2) calls (1)
+	store42Program := program.New().Sstore(0x42, 0x42)
+	callBobProgram := program.New().Call(nil, bobAddr, 1, 0, 0, 0, 0)
+
+	alloc := *actionsHelpers.DefaultAlloc
+	alloc.L2Alloc = make(map[common.Address]types.Account)
+	alloc.L2Alloc[aa] = types.Account{
+		Code: store42Program.Bytes(),
+	}
+	alloc.L2Alloc[bb] = types.Account{
+		Code: callBobProgram.Bytes(),
+	}
+
+	testCfg.Allocs = &alloc
+
+	tp := helpers.NewTestParams()
+	cfg := helpers.NewBatcherCfg()
+	env := helpers.NewL2FaultProofEnv(t, testCfg, tp, cfg)
+	sequencer := env.Sequencer
+	miner := env.Miner
+	batcher := env.Batcher
+
+	require.Equal(gt, env.Bob.Address(), bobAddr)
+	u1 := sequencer.L2Unsafe()
+
+	sequencer.ActL2EmptyBlock(t) // we'll inject the setcode tx in this block's batch
+
+	aliceSecret := env.Alice.L2.Secret()
+
+	chainID := env.Sequencer.RollupCfg.L2ChainID
+
+	// Sign authorization tuples.
+	// The way the auths are combined, it becomes
+	// 1. tx -> addr1 which is delegated to 0xaaaa
+	// 2. addr1:0xaaaa calls into addr2:0xbbbb
+	// 3. addr2:0xbbbb  writes to storage
+	auth1, err := types.SignSetCode(aliceSecret, types.SetCodeAuthorization{
+		ChainID: *uint256.MustFromBig(chainID),
+		Address: bb,
+		Nonce:   0,
+	})
+	require.NoError(gt, err, "failed to sign auth1")
+
+	txdata := &types.SetCodeTx{
+		ChainID:   uint256.MustFromBig(chainID),
+		Nonce:     0,
+		To:        env.Alice.Address(),
+		Gas:       500000,
+		GasFeeCap: uint256.NewInt(5000000000),
+		GasTipCap: uint256.NewInt(2),
+		AuthList:  []types.SetCodeAuthorization{auth1},
+	}
+	signer := types.NewIsthmusSigner(chainID)
+	tx := types.MustSignNewTx(aliceSecret, signer, txdata)
+
+	txdata2 := &types.DynamicFeeTx{
+		ChainID:   chainID,
+		Nonce:     1,
+		To:        nil,
+		Gas:       500000,
+		GasFeeCap: big.NewInt(5000000000),
+		GasTipCap: big.NewInt(2),
+	}
+	tx2 := types.MustSignNewTx(aliceSecret, signer, txdata2)
+
+	batcher.ActL2BatchBuffer(t, func(block *types.Block) *types.Block {
+		// inject user tx into upgrade batch
+		return block.WithBody(types.Body{Transactions: append(block.Transactions(), tx, tx2)})
+	})
```

Comment:
If what we care about is coverage of the span batch handling of forbidden creation bits, couldn't we just add two dummies here? Since we'd assert on the right log message, we'd know that we cover the right error path.

Reply:
Cleaned this up here: f6fb563